src/measurements.rs

```diff
@@ -146,15 +161,12 @@ fn is_valid(idx: &[usize; 2], coords: &[isize; 2], dims: &[isize; 2]) -> Option<
     let valid = |i, c, d| -> Option<usize> {
         let a = i as isize + (c - 1);
         if a >= 0 && a < d {
-            if c == 0 {
-                return Some(i - 1);
-            }
+            Some(a as usize)
         } else {
-            return None;
+            None
         }
-        Some(i)
```

Comment:
TBH, for this function, I simply copied what SciPy did without asking too much questions :) but I'm surprised by those changes. (I'm not saying they are right or wrong).

Can you please explain why you changed this? Was there a bug before? It seems strange to me that it was working with the star kernel before.

Reply:
The closest line from scipy is https://github.com/scipy/scipy/blob/v0.16.1/scipy/ndimage/src/_ni_label.pyx#L341, the resulting neighbor coordinate should be `i + c - 1` (or for scipy `ito.coordinates[idim] + itstruct.coordinates[idim] - 1`). While this function was returning `if c == 0 { i - 1} else { i }`. This was fine previously, because for the star kernel (after the all-false rows are removed), c can only be 0 or 1. But when you use more connected kernels, c can also equal 2 and then it went wrong.

Reply:
Right, thank you for the explanation.